tensorflow/contrib/seq2seq/python/ops/decoder_fn.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2016 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+"""Seq2seq loss operations for use in neural networks.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from tensorflow.python.ops import array_ops
+
+
+__all__ = ["decoder_fn_train",
+           "decoder_fn_eval_wrapper"]
+
+def decoder_fn_train(encoder_state):
```

Comment:
also consider merging decoder_fn_train and decoder_fn_eval into `get_simple_argmax_decoder_fn` and select which to return based on a boolean argument `is_training`.